src/Gemstone.Web/WebExtensions.cs

```diff
@@ -37,13 +46,185 @@ public static class WebExtensions
+        /// <param name="result">HttpResponse from controller.</param>
+        /// <param name="message">Response message from .NET Framework App.</param>
+        /// <param name="cancellationToken">Token to cancel the action.</param>
+        public static async Task SetValues(this HttpResponse result, HttpResponseMessage? message, CancellationToken cancellationToken) =>
+            await result.SetValues(message, null, null, cancellationToken);
+
+
+        /// <summary>
+        /// Used to load headers, response phrase, and status code into <see cref="HttpResponse"/> for use of proxy endpoints between .NET Core and .NET Framework Apps.
+        /// </summary>
+        /// <param name="result">HttpResponse from controller.</param>
+        /// <param name="message">Response message from .NET Framework App.</param>
+        /// <param name="excludedHeaders">Additional headers to exclude on the copy over between objects.</param>
+        /// <param name="cookieCallback">Callback to append additional cookies to response header.</param>
+        /// <param name="cancellationToken">Token to cancel the action.</param>
+        public static async Task SetValues(this HttpResponse result, HttpResponseMessage? message, IEnumerable<string>? excludedHeaders, Action<IResponseCookies>? cookieCallback, CancellationToken cancellationToken)
+        {
+            if (message is null)
+            {
+                result.StatusCode = (int)HttpStatusCode.ServiceUnavailable;
+                return;
+            }
+
+            result.StatusCode = (int)message.StatusCode;
+            IHttpResponseFeature? feature = result.HttpContext.Response.HttpContext.Features.Get<IHttpResponseFeature>();
+            if (feature is not null)
+                feature.ReasonPhrase = message.ReasonPhrase;
+
+            // Handle Set-Cookie Header
+            if (message.Headers.TryGetValues("Set-Cookie", out IEnumerable<string>? setCookieHeaders))
+            {
+                // Forward any "Set-Cookie" headers from message back to response using ASP.NET Core cookie
+                foreach (string header in setCookieHeaders)
+                {
+                    // Try parsing HttpResponseMessage cookie header with an HttpResponse CookieOptions value,
+                    // as a fallback, forward raw header to avoid losing possible auth/session variables
+                    if (TryParseSetCookie(header, out string name, out string? value, out CookieOptions options))
+                        result.Cookies.Append(name, value ?? string.Empty, options);
+                    else
+                        result.Headers.Append("Set-Cookie", header);
+                }
+            }
+
+            if (cookieCallback is not null)
+            {
+                cookieCallback(result.Cookies);
+            }
+
+            HashSet<string> fullExcludedList = (excludedHeaders is null ? excludeHeaderBase : excludeHeaderBase.Concat(excludedHeaders)).ToHashSet();
```

Comment:
Header names are case-insensitive.
```suggestion
            HashSet<string> fullExcludedList = (excludedHeaders is null ? excludeHeaderBase : excludeHeaderBase.Concat(excludedHeaders)).ToHashSet(StringComparer.OrdinalIgnoreCase);
```